File: src/main/java/com/gmail/borlandlp/minigamesdtools/activepoints/reaction/custom/ItemGiveReaction.java
package com.gmail.borlandlp.minigamesdtools.activepoints.reaction.custom;

import com.gmail.borlandlp.minigamesdtools.Debug;
import com.gmail.borlandlp.minigamesdtools.MinigamesDTools;
import com.gmail.borlandlp.minigamesdtools.activepoints.reaction.Reaction;
import org.bukkit.entity.Entity;

public class ItemGiveReaction extends Reaction {
    @Override
    public void performDamage(Entity reactionInitiator, double damage) {
        Debug.print(Debug.LEVEL.NOTICE, "ItemGiveReaction->performDamage(" + reactionInitiator.getName() + ", " + damage + ")");
    }

    @Override
    public void performIntersection(Entity reactionInitiator) {
        Debug.print(Debug.LEVEL.NOTICE, "ItemGiveReaction->performIntersection(" + reactionInitiator.getName() + ")");
    }

    @Override
    public void performInteraction(Entity reactionInitiator) {
        Debug.print(Debug.LEVEL.NOTICE, "ItemGiveReaction->performInteraction(" + reactionInitiator.getName() + ")");
    }

    @Override
    public String toString() {
        return "{Reaction type=ItemGive}";
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/activepoints/reaction/custom/ItemGiveReaction.kt
package com.gmail.borlandlp.minigamesdtools.activepoints.reaction.custom

import com.gmail.borlandlp.minigamesdtools.Debug
import com.gmail.borlandlp.minigamesdtools.Debug.print
import com.gmail.borlandlp.minigamesdtools.activepoints.reaction.Reaction
import org.bukkit.entity.Entity

class ItemGiveReaction : Reaction() {
    override fun performDamage(reactionInitiator: Entity, damage: Double) {
        print(
            Debug.LEVEL.NOTICE,
            "ItemGiveReaction->performDamage(" + reactionInitiator.name + ", " + damage + ")"
        )
    }

    override fun performIntersection(reactionInitiator: Entity) {
        print(
            Debug.LEVEL.NOTICE,
            "ItemGiveReaction->performIntersection(" + reactionInitiator.name + ")"
        )
    }

    override fun performInteraction(reactionInitiator: Entity) {
        print(
            Debug.LEVEL.NOTICE,
            "ItemGiveReaction->performInteraction(" + reactionInitiator.name + ")"
        )
    }

    override fun toString(): String {
        return "{Reaction type=ItemGive}"
    }
}